### PR TITLE
OUT 991 When I have the formatting menu open and hit 'Escape' I want to close the formatting menu instead of being taken back to the all tasks views

### PR DESCRIPTION
**Pull request by aatbip**
### Changes

- [x] Adds a check if formatting menu is active in the dom before closing the modal

### Testing Criteria

[Screencast from 12-11-24 12:47:18 PM +0545.webm](https://github.com/user-attachments/assets/17ab9836-c74c-4283-9fde-8eb03f116684)

### CODE REVIEW

**Review by rrojan — CHANGES_REQUESTED** [review]
> I want to close the formatting menu instead of being taken back to the all tasks views

The main issue here still persists.

This is expected to work in both task create modal, as well as task description.

[Screencast from 2024-11-12 13-00-03.webm](https://github.com/user-attachments/assets/dd5f5d8e-17ef-4d75-85de-138a447d1138)

File: src/app/ui/Modal_NewTaskForm.tsx [hunk 1]
```diff
@@ -29,16 +29,24 @@ export const ModalNewTaskForm = ({
...
+  const handleModalClose = async ({ triggeredFrom }: { triggeredFrom: 'MOUSE' | 'KEYBOARD' }) => {
+    const hasSlashCommandMenu = document.querySelector('.tippy-box') !== null
+    if (!(triggeredFrom === 'KEYBOARD' && hasSlashCommandMenu)) {
+      store.dispatch(setShowModal())
+      store.dispatch(clearCreateTaskFields({ isFilterOn: !!filterOptions[FilterOptions.ASSIGNEE] }))
+      store.dispatch(setActiveWorkflowStateId(null))
+    }
     // NOTE: Reimplement in M3
     // await bulkRemoveAttachments(attachments)
   }
```
Comment (priosshrsth) [inline src/app/ui/Modal_NewTaskForm.tsx:41]:
```suggestion
    if (triggeredFrom === 'KEYBOARD' && document.querySelector('.tippy-box')) {
      return;
    }
    
    store.dispatch(setShowModal())
      store.dispatch(clearCreateTaskFields({ isFilterOn: !!filterOptions[FilterOptions.ASSIGNEE] }))
      store.dispatch(setActiveWorkflowStateId(null))
    /* TODO: Reimplement in M3
    await bulkRemoveAttachments(attachments)
    */
  }
```

File: src/app/ui/NewTaskForm.tsx [hunk 2]
```diff
@@ -107,7 +107,7 @@ export const NewTaskForm = ({ handleCreate, handleClose, getSignedUrlUpload }: N
   useEffect(() => {
     function handleEscPress(e: KeyboardEvent) {
       if (e.key === 'Escape') {
-        handleClose()
+        handleClose({ triggeredFrom: 'MOUSE' })
```
Comment (rrojan) [inline src/app/ui/NewTaskForm.tsx:110]: Instead of adding `triggeredFrom: 'Mouse'` to every button and icon, we can make it the default behavior

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/app/ui/NewTaskForm.tsx (hunk 2 above)
Comment (aatbip) [reply src/app/ui/NewTaskForm.tsx:110]: Using a default value for `triggeredFrom` can lead to implicit behavior, making it unclear what triggered the modal close if `triggeredFrom` is omitted by mistake. Explicitly specifying 'MOUSE' or 'KEYBOARD' keeps the intent clear and improves readability. This approach avoids potential debugging issues due to unintended default behavior.

Comment (priosshrsth) [reply src/app/ui/NewTaskForm.tsx:110]: @aatbip I think we only care when trigged  by keyboard, isn't it? In that case, we could make it simpler by making this optional parameter. Which should've kept the changes minimal. But now that you have made the changes, I do agree with you here that it keeps intention clear. But the way it is done, looks bothersome to provide the triggeredFrom params everywhere.

Comment (aatbip) [reply src/app/ui/NewTaskForm.tsx]: Changes added in the commit below @rrojan @priosshrsth

**Review by rrojan — CHANGES_REQUESTED** [review]
@aatbip When i add some changes, then a command like `/up` and press escape to close the formatting menu the changes are lost...

[Screencast from 2024-11-12 19-25-39.webm](https://github.com/user-attachments/assets/d076a8e6-58d9-4f9f-a187-02a649d5c935)

**Comment by rrojan** [comment]
@aatbip It seems like Realtime is having issues on Supabase again. Is it possible this is the cause? After this is resolved please look through this once again.

image

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Pull request merged**